AirCasting/ABConnector/SDCardAirBeamServices.swift

```diff
@@ -210,13 +209,17 @@ class BluetoothSDCardAirBeamServices: SDCardAirBeamServices, BluetoothConnection
             self.finishSync(device: device) { completion(.failure(SDCardSyncError.wrongOrderOfReceivedPayload)) }
             return
         }
-        self.receivedMeasurementsCount[sessionType, default: 0] += Constants.SDCardSync.numberOfMeasurementsInDataChunk
+        
+        
+        let numberOfMeasurementsInChunk =  payload.components(separatedBy: "\r\n").filter { !$0.trimmingCharacters(in: ["\n"]).isEmpty }.count
```

Comment:
Measurements aren't ALWAYS coming in fours, so this is taking into account the fact that there might not be full 4 measurements in the data pack.